sw/device/lib/dif/dif_uart.h

```diff
@@ -158,7 +159,8 @@ bool dif_uart_bytes_send(const dif_uart_t *uart, const uint8_t *data,
  *
  * Non-blocking UART receive bytes routine. Can be used from inside an UART ISR.
  * This function attempts to read @p arg2 number of bytes from the UART RX FIFO
- * into @p arg3, and passes @p arg4 back to the caller.
+ * into @p arg3, and passes @p arg4 back to the caller. @p arg4 is optional,
+ * NULL should be passed in if the value is not needed.
```

Comment:
Where is the `@p argN` syntax coming from? `@p` is doxygen, but it expects the name of the argument as parameter, not `argN`. (http://www.doxygen.nl/manual/commands.html#cmdp).

Reply:
I admit I haven't looked in doxygen documentation, instead I just copied what the coding style of [OT](https://docs.opentitan.org/doc/rm/c_cpp_coding_style/) says:
```
/**
 * Do something amazing
 *
 * Create a rainbow and place a unicorn at the bottom of it. @p arg1 pots of
 * gold will be positioned on the east end of the rainbow.
 *
 * @param pots_of_gold Number of gold pots to place next to the rainbow
 * @param unicorns Number of unicorns to position on the rainbow
 * @return 0 if the function was successful, -1 otherwise
 */
int create_rainbow(int pots_of_gold, int unicorns);
```

I get now what it was trying to show, but I think it should be changed :)

Reply:
I can just change both PLIC and UART dif function descriptions in a separate PR?

Reply:
yes, please do.